archaius-core/src/main/java/com/netflix/config/DynamicBooleanProperty.java

```diff
@@ -23,15 +23,42 @@
  *
  */
 public class DynamicBooleanProperty extends PropertyWrapper<Boolean> {
+
+    protected volatile boolean primitiveValue;
+
     public DynamicBooleanProperty(String propName, boolean defaultValue) {
         super(propName, Boolean.valueOf(defaultValue));
+
+        // Set the initial value of the cached primitive value.
+        this.primitiveValue = chooseValue();
+    }
+
+    @Override
+    protected void propertyChanged() {
+        super.propertyChanged();
```

Comment:
I'm a little wary of calling super here. It's a no-op there at super, so overhead is less of concern. But it maybe confusing for people down the road who might expect if they make the change in super, it would take effect in subclasses as well.